get: add example on downloading normal git files (#821)
iterative/dvc.org: static/docs/command-reference/get.md

# get

Download a file or directory from any <abbr>DVC project</abbr> or Git repository
(e.g. hosted on GitHub) into the current working directory.

> Unlike `dvc import`, this command does not track the downloaded files (does
> not create a DVC-file).

## Synopsis

```usage
usage: dvc get [-h] [-q | -v] [-o [OUT]] [--rev [REV]] url path

Download files or directories from DVC repository.
Documentation: <https://man.dvc.org/get>

> 

Reviewer: We don't add these to the usage blocks in command references because it's redundant 🙂

Reviewer: See https://github.com/iterative/dvc.org/pull/821#discussion_r350966417

positional arguments:
  url                   URL of Git repository with DVC project to
                        download from.
  path                  Path to a file or directory within a
                        DVC repository.

optional arguments:
  -h, --help            show this help message and exit
  -q, --quiet           Be quiet.
  -v, --verbose         Be verbose.
  -o [OUT], --out [OUT]
                        Destination path.

>   --rev [REV]           DVC repository git revision.

Reviewer: Same with all this. We edit for better looking here. And no need to include the options since they are described in detail later in this doc. Let's revert this whole chunk please.

>   --rev [REV]           DVC repository git revision.

Reviewer: Same with all this. We edit for better looking in the cmd ref. And no need to include the options since they are described in detail later in this doc. Let's revert this whole chunk please.

```

## Description

Provides an easy way to download datasets, intermediate results, ML models, or

> other files and directories tracked in another <abbr>DVC repository</abbr>, by

Reviewer: Why are we removing `(any <abbr>data artifact</abbr>)` ?

Reviewer: Oh I see. This is for iterative/dvc/pull/2837.

online repositories (e.g. `[user@]server:project.git`). `url` can also be a
local file system path to an "offline" repository.

The `path` argument of this command is used to specify the location of the file
or directory within the source project. If the file is a
[DVC-file](/doc/user-guide/dvc-file-format) the source project must have a
default [DVC remote](/doc/command-reference/remote) configured.

> > See `dvc get-url` to obtain data from other supported URLs.

Reviewer: In this case `dvc get-url` has nothing to do with Git so "download" is correct and precise here, I think.

- `-v`, `--verbose` - displays detailed tracing information.

> ## Example: Retrieve a model from a DVC remote

Reviewer: > from a DVC remote

To the user this may not be obvious. They don't indicate the remote but the URL (or path) to the DVC repo. So maybe say "from an external DVC repository" or just leave "Examples". Open to suggestions.

Reviewer: How about:

```suggestion
## Example: Download a file tacked by an external DVC repository
```

> Note that `dvc get` can be used from anywhere in the file system, as long as
> DVC is [installed](/doc/install).

We can use `dvc get` to obtain the resulting model file from our

> [get started example repo](https://github.com/iterative/example-get-started), a

Reviewer: In this example also we're talking about a remote URL, so the term "download" was correct.

Reviewer: Sorry, I know it may sound like I'm contradicting myself in https://github.com/iterative/dvc.org/pull/821/#discussion_r351033343, but I didn't mean to just replace all instances of the word 🙂

its
[config file](https://github.com/iterative/example-get-started/blob/master/.dvc/config)).

> > A recommended use for obtaining binary files from DVC repositories, as done in

Reviewer: "downloading" was OK for the same reason as my previous comment.

> this example, is to place a ML model inside a wrapper application that serves
> as an [ETL](https://en.wikipedia.org/wiki/Extract,_transform,_load) pipeline
> or as an HTTP/RESTful API (web service) that provides predictions upon
> request. This can be automated leveraging DVC with
> [CI/CD](https://en.wikipedia.org/wiki/CI/CD) tools.

The same example applies to raw or intermediate <abbr>data artifacts</abbr> as
well, of course, for cases where we want to obtain those files or directories
and perform some analysis on them.

> ## Examples: Retrieve a file from a git repository

Reviewer: ```suggestion
## Examples: Download a file tracked by a Git repository
```

We can also use `dvc get` to retrieve any file or directory that exists in a git
repository.

```dvc
$ dvc get https://github.com/schacon/cowsay/install.sh install.sh
$ ls
install.sh
```

## Example: Compare different versions of data or model

`dvc get` has the `--rev` option, to specify which version of the repository to
obtain a <abbr>data artifact</abbr> from. It also has the `--out` option to

> specify the target path. Combining these two options allows us to do something

Reviewer: Thanks. I agree "target path" is better than before but still a little ambiguous. How about something like "location to place the artifact in the <abbr>workspace</abbr>"?

`-o model.bigrams.pkl` and `--rev 9-bigrams-model` or even without `--rev`
(since it's the latest version anyway). In fact, in this case using `dvc pull`
with the corresponding [DVC-files](/doc/user-guide/dvc-file-format) should

> suffice, obtaining the file as just `model.pkl`. We can then rename it to make

Reviewer: Again, in this particular example we're actually downloading :)
